src/util.c

```diff
@@ -633,3 +633,62 @@ All facets connected.  No further nearby check necessary.\n");
     stl_verify_neighbors(stl);
   }
 }
+
+stl_file *stl_copy(stl_file *dst, const stl_file *src){
+
+  if(src->error) return NULL;
+
+  dst->fp               = NULL;
+  dst->facet_start      = NULL;
+  dst->edge_start       = NULL;       // edge_start is never allocated nor populated.
+  dst->heads            = NULL;       // Allocated and used only by stl_fill_holes and immediately freed.
+  dst->tail             = NULL;       // Same as ->heads.
+  dst->M                = src->M;
+  dst->neighbors_start  = NULL;
+  dst->v_indices        = NULL;
+  dst->v_shared         = NULL;
+  dst->stats            = src->stats;
+  dst->stats.type       = inmemory;
+  dst->error            = src->error;
```

Comment:
(bikeshedding) Would this make more sense as an explicit non-error?

```suggestion
  dst->error            = 0;
```